deckhouse-controller/pkg/controller/deckhouse-release/updater/updater.go

```diff
@@ -160,3 +183,23 @@ func (api *KubeAPI) SaveReleaseData(ctx context.Context, release *v1alpha1.Deckh
 		NotifiedAnnotation:   strconv.FormatBool(data.Notified),
 	})
 }
+
+func (api *KubeAPI) IsKubernetesVersionAutomatic(ctx context.Context) (bool, error) {
+	key := client.ObjectKey{Namespace: "kube-system", Name: "d8-cluster-configuration"}
+	secret := new(corev1.Secret)
+	if err := api.client.Get(ctx, key, secret); err != nil {
+		return false, fmt.Errorf("check kubernetes version: failed to get secret: %s", err.Error())
+	}
+
+	var clusterConf struct {
+		KubernetesVersion string `json:"kubernetesVersion"`
+	}
+	clusterConfigurationRaw, ok := secret.Data["cluster-configuration.yaml"]
+	if !ok {
+		return false, fmt.Errorf("check kubernetes version: expected field 'cluster-configuration.yaml' not found in secret %s", secret.Name)
+	}
+	if err := yaml.Unmarshal(clusterConfigurationRaw, &clusterConf); err != nil {
+		return false, fmt.Errorf("check kubernetes version: failed to unmarshal cluster configuration: %s", err)
+	}
+	return clusterConf.KubernetesVersion == "Automatic", nil
```

Comment:
"Automatic" not const. we use it here only?
maybe it will be better to use constant value?

Reply:
fixed, but it's going to take time and efforts to replace all ad-hoc strings in the project